src/modules/accommodations/accommodations.service.ts

```diff
@@ -0,0 +1,55 @@
+import { Injectable } from '@nestjs/common';
+import { Accommodation } from './entities';
+import { Repository } from 'typeorm';
+import { InjectRepository } from '@nestjs/typeorm';
+import { SearchAccommodationParamsDto } from './dto/request';
+import { PaginatedResult } from './interfaces';
+
+@Injectable()
+export class AccommodationsService {
+  constructor(
+    @InjectRepository(Accommodation)
+    private readonly accommodationRepository: Repository<Accommodation>,
+  ) {}
+
+  async getAccommodations(
```

Comment:
I think you can also return amenities and category of the accommodation.

Reply:
right now it returns category only when the query param is passed

Reply:
because on the frontend they know which category they are searching for and amenities are not used in homepage